src/sign.ml

```diff
@@ -277,6 +282,29 @@ let add_rule : t -> sym -> rule -> unit = fun sign sym r ->
     let m = (sym.sym_name, r) :: m in
     sign.sign_deps := PathMap.add sym.sym_path m !(sign.sign_deps)
 
+(** [add_rules s r] adds a bunch of rules [r] to its attached symbol to the
+    signature [s] and builds decision trees. *)
+let add_rules : t -> (sym * pp_hint * rule loc) list -> unit = fun sign rs ->
+  let add_rule (s, h, r) =
+    out 3 "(rule) %a\n" Print.pp_rule (s,h,r.elt) ;
+    add_rule sign s r.elt in
+  List.iter add_rule rs ;
```

Comment:
This is not very efficient as add_rule is doing https://github.com/Deducteam/lambdapi/blob/33809c57b1b39aeb65bb0583b2139bdec46b2e4e/src/sign.ml#L271. These two functions should be merged. We should only have a add_rules function.

Reply:
By the way, since we don't care about the order of the rules, we may not have to perform this concatenation, we might as well do a cons, right? (`r :: !sym.sym_rules`)

Reply:
Right! But it may still be useful to merge these 2 functions to factorize some code.

Reply:
Indeed, indeed, it has been done in a920949